src/blockchain/transactions.ts

```diff
@@ -243,3 +253,12 @@ export const transactions$: Observable<TxState[]> = combineLatest(
   startWith([]),
   shareReplay(1),
 );
+
+interface ExternalNonce2tx { [nonce: number]: { hash: string, callData: string }; }
+const externalNonce2tx$: Observable<ExternalNonce2tx> = combineLatest(context$, account$, onEveryBlock$.pipe(first()), onEveryBlock$).pipe(
+  switchMap(([context, account, firstBlock]) => ajax({ url: `${context.etherscan.apiUrl}?module=account&action=txlist&address=${account}&startblock=${firstBlock}&sort=desc&apikey=${context.etherscan.apiKey}` })),
+  map(({ response }) => response.result),
+  map((transactions: Array<{ hash: string, nonce: string, input: string }>) => fromPairs(transactions.map<[string, { hash: string, callData: string }]>(tx => [tx.nonce, { hash: tx.hash, callData: tx.input }]))),
```

Comment:
I guess this can be extracted as type because you used the same `interface` in the `send` function